src/fpm_targets.f90

```diff
@@ -307,6 +307,17 @@ subroutine build_target_list(targets,model)
                                     output_name = join_path(exe_dir, &
                                     sources(i)%exe_name//xsuffix))
 
+                    ! If the main program is on a C/C++ source, the Intel Fortran compiler requires option
+                    ! -nofor-main to avoid "duplicate main" errors.
+                    ! https://stackoverflow.com/questions/36221612/p3dfft-compilation-ifort-compiler-error-multiple-definiton-of-main
+                    if (model%compiler%is_intel() .and. any(exe_type==[FPM_TARGET_C_OBJECT,FPM_TARGET_CPP_OBJECT])) then
+                       if (get_os_type()==OS_WINDOWS) then
+                           targets(size(targets))%ptr%compile_flags = '/nofor-main'
+                       else
+                           targets(size(targets))%ptr%compile_flags = '-nofor-main'
+                       end if
+                    end if
```

Comment:
Can we put this in the compiler type to provide us with extra link / compile flags?

Reply:
I was trying to do it but unfortunately after that location the link between the `.exe` target and the fact that it comes from a C/C++ source is lost (two separate targets are made: the object and the executable). 

They only need to apply to the linker command when the main program is from a C/C++ source, so, not necessarily general.

Maybe I could add a "main-specific flags" wrapper in the compiler object, to at least hide the Intel-only hack?